beamer/tests/agent/multiple_processors/test_multiple_event_processors.py

```diff
@@ -0,0 +1,241 @@
+        "brownie",
+        "--chain.chainId",
+        str(chain_id),
+    ]
+    return psutil.Popen(cmd, stdin=DEVNULL, stdout=PIPE, stderr=PIPE)
+
+
+def _start_ganache(chain_id: ChainId, port: str) -> psutil.Popen:
+    return _start_ganache_process(chain_id, port)
+
+
+def _kill_ganache(process: psutil.Popen):
+    os.kill(process.pid, signal.SIGKILL)
+    process.wait()
+
+
+def _start_slave_test(port: int, request_count: int) -> psutil.Popen:
+    _create_new_config(port)
+    return psutil.Popen(
+        ["ape", "test", str(SLAVE_TEST_PATH), "-s"],
+        bufsize=0,
+        stdin=PIPE,
+        stdout=PIPE,
+        stderr=PIPE,
+        env=dict(os.environ, PORT=str(port), REQUEST_COUNT=str(request_count)),
+    )
+
+
+@contextlib.contextmanager
+def _new_networks(chain_port_map):
+    processes = {}
+    try:
+        current_config = _create_and_return_backup_config()
+        for chain_id, port in chain_port_map.items():
+            processes[chain_id] = _start_ganache(chain_id, port)
+        yield current_config
+    finally:
+        for process in processes.values():
+            _kill_ganache(process)
+
+
+def _get_chain_map() -> dict[ChainId, int]:
+    number_of_chains = 4
+    return {ChainId(i): 8545 + i for i in range(1, number_of_chains + 1)}
+
+
+def _get_config(
+    chain_map: dict[ChainId, int],
+    contracts,
+    allowance,
+    slave_contract_addresses: dict[ChainId, dict[str, str]],
+) -> Config:
+    account = eth_account.Account.from_key(_LOCAL_ACCOUNT.private_key)
+    url = ape.config.provider.uri
+    deployment_info = {}
+
+    for chain_id in chain_map:
+        contracts_info = dict(
+            RequestManager=ContractInfo(
+                deployment_block=BlockNumber(1),
+                address=to_canonical_address(
+                    slave_contract_addresses[chain_id]["request_manager"]
+                ),
+                abi=[abi.dict() for abi in contracts.request_manager.contract_type.abi],
+            ),
+            FillManager=ContractInfo(
+                deployment_block=BlockNumber(1),
+                address=to_canonical_address(slave_contract_addresses[chain_id]["fill_manager"]),
+                abi=[abi.dict() for abi in contracts.fill_manager.contract_type.abi],
+            ),
+        )
+        deployment_info[chain_id] = contracts_info
+    rpc_urls = {"l1": url}
+    for chain_id, port in chain_map.items():
+        rpc_url = f"http://127.0.0.1:{port}"
+        rpc_urls[str(chain_id)] = rpc_url
+
+    token_list = _get_token_list(chain_map, allowance, slave_contract_addresses)
+    config = Config(
+        rpc_urls=rpc_urls,
+        deployment_info=deployment_info,
+        token_checker=TokenChecker(token_list),
+        account=account,
+        fill_wait_time=0,
+        unsafe_fill_time=600,
+        prometheus_metrics_port=None,
+        log_level="debug",
+    )
+    beamer.agent.metrics.init(config=config, source_rpc_url=url, target_rpc_url=url)
```

Comment:
Why do we need this here?

Reply:
it is there also for conftest config fixture. i can just delete it

Reply:
> it is there also for conftest config fixture. i can just delete it

I seem to remember that the metrics are being used internally so that may be the reason we first need to initialize that module. Could you please check that?

Reply:
in agent.py we are initializing metrics already in start, maybe this is not necessary.